gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -185,39 +184,35 @@ private ManagedChannel createSingleChannel() throws IOException {
     String serviceAddress = endpoint.substring(0, colon);
 
     // TODO(hzyi): Use NettyChannelBuilder when maxHeaderListSize is specified to unblock Spanner.
-    // Changed to ManagedChannelBuilder when https://github.com/grpc/grpc-java/issues/4050 is resolved.
+    // Changed to ManagedChannelBuilder when https://github.com/grpc/grpc-java/issues/4050 is
+    // resolved.
+    ManagedChannelBuilder builder;
     if (maxHeaderListSize != null) {
-      NettyChannelBuilder builder =
-          NettyChannelBuilder.forAddress(serviceAddress, port)
-              .intercept(headerInterceptor)
-              .intercept(metadataHandlerInterceptor)
-              .userAgent(headerInterceptor.getUserAgentHeader())
-              .executor(executor)
-              .maxHeaderListSize(maxHeaderListSize);
-      if (maxInboundMessageSize != null) {
-        builder.maxInboundMessageSize(maxInboundMessageSize);
+      try {
+        Class.forName("io.grpc.netty.NettyChannelBuilder");
+        builder =
+            io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder.forAddress(serviceAddress, port);
+      } catch (ClassNotFoundException e) {
+        try {
+          Class.forName("io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder");
+          builder = io.grpc.netty.NettyChannelBuilder.forAddress(serviceAddress, port);
+        } catch (ClassNotFoundException e) {
+          throw new RuntimeException(
+              "Unable to create the channel because neither"
+              + " \"io.grpc.netty.NettyChannelBuilder\" nor"
+              + " \"io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder\" is found.");
+        }
       }
-      if (keepAliveTime != null) {
-        builder.keepAliveTime(keepAliveTime.toMillis(), TimeUnit.MILLISECONDS);
-      }
-      if (keepAliveTimeout != null) {
-        builder.keepAliveTimeout(keepAliveTimeout.toMillis(), TimeUnit.MILLISECONDS);
-      }
-      if (keepAliveWithoutCalls != null) {
-        builder.keepAliveWithoutCalls(keepAliveWithoutCalls);
-      }
-      if (interceptorProvider != null) {
-        builder.intercept(interceptorProvider.getInterceptors());
-      }
-      return builder.build();
+      builder.maxHeaderListSize(maxHeaderListSize);
```

Comment:
This is a compile error (You can see it in the Travis build) - the type here is `ManagedChannelBuilder`. You need to call it while you still have a reference to the Netty builder.

Reply:
Sorry I forgot to build first :/ I'll keep working on it.